etna/records/tests/test_views.py

```diff
@@ -722,3 +723,72 @@ def test_invalid_response_from_kong_raises_404(self):
         response = self.client.get("/records/images/C123456/11000000000000000000/")
 
         self.assertEquals(response.status_code, 404)
+
+
+class RecordDetailBackToSearchTest(TestCase):
+    @responses.activate
+    def test_back_to_search_context_with_catalogue_search(self):
+        expected_url = "/search/catalogue/?sort_by=title&q=london&filter_keyword=paper&level=Item&collection=ADM&collection=BT&closure=Open+Document%2C+Open+Description&opening_start_date_0=&opening_start_date_1=&opening_start_date_2=1900&opening_end_date_0=&opening_end_date_1=&opening_end_date_2=2020&per_page=20&sort_order=asc&display=list&page=2&group=tna"
+
+        responses.add(
+            responses.GET,
+            "https://kong.test/data/fetch",
+            json=create_response(
+                records=[
+                    create_record(iaid="C13359805"),
+                ]
+            ),
+        )
+        record_detail_url = reverse(
+            "details-page-machine-readable", kwargs={"iaid": "C13359805"}
+        )
```

Comment:
We should move as much duplicate code from out of individual test cases and into `setUp()` as we can